### PR TITLE
[v2r4] Openstack and other fixes

### PR DESCRIPTION
**Pull request by atsareg**
Several fixes as requested in the [forum](https://groups.google.com/forum/#!topic/diracgrid-forum/nghBih6Y7KA)

BEGINRELEASENOTES

*Resources
FIX: Openstack - NumberOfProcessors is taken from the flavor definition
FIX: cloudinit template defines VMDIRAC_VERSION and JOB_ID environment

*WMS
FIX: VirtualMachineCLI - enhanced interface and doc strings

ENDRELEASENOTES

### CODE REVIEW

File: VMDIRAC/Resources/Cloud/OpenStackEndpoint.py [hunk 1]
```diff
@@ -206,7 +206,10 @@ def createInstance(self, instanceID=''):
       flavorID = self.flavors.get(flavor)["FlavorID"]
       if not flavorID:
         return S_ERROR('Can not get ID for the flavor: %s' % flavor)
+      numberOfProcessors = self.flavors.get(flavor)["NumberOfProcessors"]
     self.parameters['FlavorID'] = flavorID
+    if "NumberOfProcessors" not in self.parameters:
+      self.parameters["NumberOfProcessors"] = numberOfProcessors
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenStackEndpoint.py:212]: I know pretty much nothing about VMDIRAC, so forgive me if I write silly stuff, but...
IIUC this "NumberOfProcessors" is the number of processors of the VMs that VMDIRAC will try to instantiate. I see this is going in the OpenStack endpoint module, but I think it should be common to all the endpoints, so going in https://github.com/DIRACGrid/VMDIRAC/blob/integration/VMDIRAC/Resources/Cloud/CloudEndpoint.py ?

**Comment by atsareg** [comment]
@fstagni: You are right that this should be the same for all the Endpoint types. Actually, it used to be NumberOfCPUs parameter. I have changed it to be NumberOfProcessors everywhere to be in sync with DIRAC. Now NumberOfProcessors is defined for all the cloud types. For most of them it is derived from the cloud info, but if it can not be obtained from the cloud endpoint, it can defined in the CS in the VMType definition

**Comment by sfayer** [comment]
Hi,
It seems that about half of the cloud endpoints default to 1 processor per instance, whereas the others default to 2. Perhaps it would be possible to make this consistent across all of them? Otherwise, it all looks fine to me.
Regards,
Simon

**Review by sfayer — APPROVED** [review]

**Comment by atsareg** [comment]
Set NumberOfProcessors to 1 everywhere

**Pull request merged**